scripts/apps/search/directives/ItemPreview.js

```diff
@@ -45,6 +45,14 @@ export function ItemPreview(asset, storage) {
 
             scope.$watch('item', (item) => {
                 scope.selected = {preview: item || null};
+
+                // Set the desk and stage names
+                if (item && item.task && item.task.stage) {
+                    scope.deskName = desks.deskLookup[item.task.desk].name;
+                    scope.stage = desks.stageLookup[item.task.stage];
```

Comment:
A minor point: if you could do `scope.stage = desks.stageLookup[item.task.stage].name` here then you need only one `<span>` in preview.html (the 2nd one).